javascripts/discourse/initializers/upload-consent-initializer.js

```diff
@@ -1,38 +1,36 @@
 import { withPluginApi } from "discourse/lib/plugin-api";
-import { inject as service } from "@ember/service";
 import UploadConsent from "../components/modal/upload-consent";
 
 const uploadRegexp = /\(upload?:\/\/[\w\d./?=#]+\)/;
+import { Promise } from "rsvp";
 
 function initialize(api) {
-  api.modifyClass("service:composer", {
-    pluginId: "discourse-upload-consent",
-
-    modal: service(),
-
-    save() {
-      const enabledCategories = settings.consent_enabled_categories
-        .split("|")
-        .map((id) => parseInt(id, 10));
-
-      if (
-        enabledCategories.includes(this.model.categoryId) &&
-        uploadRegexp.test(this.model.reply)
-      ) {
-        const originalSave = this._super;
-        this.modal.show(UploadConsent, {
+  const composerService = api.container.lookup("service:composer");
+  const modal = api.container.lookup("service:modal");
+
+  const enabledCategories = settings.consent_enabled_categories
+    .split("|")
+    .map((id) => parseInt(id, 10));
+
+  api.composerBeforeSave(() => {
+    const categoryId = composerService?.model?.categoryId;
+    const reply = composerService?.model?.reply;
+
+    if (!categoryId || !reply) {
+      return Promise.resolve();
```

Comment:
…so that this can be just a return:

```suggestion
      return;
```